src/main.py

```diff
@@ -2716,6 +2717,15 @@ def post_all(self, o):
         email = post["email"]
         signature = post["signature"]
         quicklinks = post["quicklinksid"]
+        rawfavouritereports = post["favouritereportsid"]
+        favouritereports = ""
+        for reportid in rawfavouritereports.split(","):
+            for report in asm3.reports.get_reports(o.dbo):
```

Comment:
this is going to open a database connection and execute the get_reports query for every single iteration of the outer loop - call it once outside the loop and assign it to a variable to iterate here instead so you aren't re-reading it from the database every time

Reply:
Sorted